app/assets/javascripts/map.js

```diff
@@ -0,0 +1,52 @@
+var mapCenter;
+  if (! searchCity){
+    var searchCity = "";
+  }
+  geocoder = new google.maps.Geocoder();
+
+  geocoder.geocode({'address': searchCity}, function(results, status){
```

Comment:
Mixed double and single quotes.<br>'geocoder' is not defined.<br>'status' is defined but never used.